austin.intermediate.summer2.hagman.manny

libersolis-dev opened the pull request:
First Commit

Commit:
First Commit

Commit:
README.md updated

Commit:
Homework Complete

Commit:
01PigLatin.js Homework Complete

Commit:
Home Complete(edited typos) 01PigLatin.js

Commit:
01PigLatin.js Complete-C

maddiesolz commented:
Great work on Homework1! I see all the tests passing. Lets take a look at 01PigLatin....

maddiesolz commented on homework/01LessonOne.js:
```diff
@@ -202,13 +209,24 @@ var stringFourStrictNotEqualsNumberFour;
+//!== does not convert type so any comparison with two different types is automatically considered not equal.
+//x !== '8' is saying that "x is not equal to '8' and is a true statement. //
 // ****
 
```
Great work on Homework1! I see all the tests passing. Lets take a look at 01PigLatin....

![pig](http://66.media.tumblr.com/tumblr_lmafhoATpo1qd4wkzo1_400.gif)

maddiesolz commented on apps/01PigLatin.js:
```diff
@@ -7,11 +7,128 @@ prompt.start();
+  //
+  // Since 'a' !== 'A' we would normally need search for the uppercase and lowercase strings.
+  //
+  // Since .toLowerCase makes every letter in a string lowercase we only need to search for each vowel one time.
```
Very good! If we had left the word as it what would be and If statement for each case of the letter.

maddiesolz commented on apps/01PigLatin.js:
```diff
@@ -7,11 +7,128 @@ prompt.start();
+  // var mySlicedName = myName.slice(1, 5) ;
+  // will return a new string 'anny'
+  //
+  // We use slice() to extract parts of one string into two different new strings so they can be concatenated into 1 new string.
```
Good answer to this! Just for future reference, in computer science it is commonly called a "substring." In other languages (like Java) it would be variable.substring() instead of .slice(). Javascript is just hipster that way.

libersolis-dev replied:
That's very good to know! So if I understand correct a substring is a string within a string. So 
"is nice" is a substring of "Ice is nice" and "Ice" is also a substring of "Ice is nice". Am I on the right path? Thanks again.

maddiesolz replied:
You got it!

paulrenenichols replied:
JavaScript is just hipster that way, @maddiesolz? lol.

maddiesolz replied:
Yeah!! @paulrenenichols  come on...'slice?' way too rad for programming

paulrenenichols commented on apps/01PigLatin.js:
```diff
@@ -7,11 +7,128 @@ prompt.start();
+  //
+  // Since 'a' !== 'A' we would normally need to search for the uppercase and lowercase strings.
+  //
+  // Since .toLowerCase transforms every letter in a string into lowercase we only need to search for each vowel one time.
```
Good job answering the question, @MannyHagman!

![seahorse](http://static.tumblr.com/b53f21170ab887e9bf7ddd668d5d66b8/7pnb9rz/xAtn3smxf/tumblr_static_8bwoeld4cicc0ksckk84sgcw8.gif)

paulrenenichols commented on apps/01PigLatin.js:
```diff
@@ -7,11 +7,128 @@ prompt.start();
+  //
+  // Your Answer:
+  //
+  // An array searc that returns a value of -1 indicates that the value being searched for cannot be found.
```
True. You can do similar things with arrays, but here we are working with strings.

paulrenenichols commented on apps/01PigLatin.js:
```diff
@@ -7,11 +7,128 @@ prompt.start();
+  //
+  // If the indexOf search returns -1 then no vowel was found.
+  //
+  // -1 helps reveal also tells us what IS able to be found. A return value > -1 indicates that a vowel exists in the string.
```
Good job!

paulrenenichols commented on apps/01PigLatin.js:
```diff
@@ -7,11 +7,128 @@ prompt.start();
+  //
+  // String.prototype.slice() extracts text from a string and with that text returns a new string. It starts to extract text from the called string from value for parameter beginSlice, and stops at the value entered for the parameter endSlice.
+  //
+  // The index for endSlice is not extracted. Any indices > endSlice are also not included in the new string.
```
Looks like you followed the link and read the documentation!

paulrenenichols commented on apps/01PigLatin.js:
```diff
@@ -7,11 +7,128 @@ prompt.start();
+//
+// prompt.get() uses the command line to prompt the user for the input needed to run the 01PigLatin.js app.
+//
+// This input becomes the initial value for the 'word' variable and this variable is called, searched, sliced and concatenated to give us the new Pig Latin string.
```
Good answer.

Commit:
02LessonTwo all tests passing

maddiesolz commented on homework/02LessonTwo.js:
```diff
@@ -126,7 +236,8 @@ function whichSpecies(character) {
-// Your Answer Goes Here:
+// Conditional statements are Javascript statements that instruct code to perform a certain action (usually return
+// value back to the function or execute the next line of code) based on the Boolean (true, false) return vale of another statement.
 //
```
Manny! Great job answering all the questions. It all looks correct to me.

libersolis-dev replied:
Yes! I am starting to feel like a 1337 Hax0r now!

Commit:
02RockPaperScissors.js complete, all tests passing.

maddiesolz commented on apps/02RockPaperScissors.js:
```diff
@@ -5,16 +5,42 @@ var prompt = require('prompt');
+    var handWin1 = 'Hand one wins!';
+    var handWin2 = 'Hand two wins!';
+    var handTie = "It's a tie!";
 
```
Good use of variables here! It will save you the trouble of having to copy/paste 'Hand one wins!" a bunch of times.

maddiesolz commented on apps/02RockPaperScissors.js:
```diff
@@ -5,16 +5,42 @@ var prompt = require('prompt');
+        return ('Invalid Input! Please enter rock, paper, or scissors to continue');
+    }
 }
 
```
Very good if/else tree! If you want this code to be EVEN MORE fun, explore the 'or' operato ( || )  in the logic as well.

Example Pseudo code: 

``` javascript
else if ( 
        (hand1==='scissors' && hand2==='paper') || 
        (hand1=== 'paper' && hand2==='rock') || 
        (hand1==='rock' && hand2==='scissors')
        )
        { 
        return "Hand one wins!"; 
        }
```

maddiesolz commented on apps/02RockPaperScissors.js:
```diff
@@ -24,13 +50,13 @@ function getPrompt() {
+        it('should detect which hand won', function() {
             assert.equal(rockPaperScissors('rock', 'paper'), "Hand two wins!");
             assert.equal(rockPaperScissors('paper', 'scissors'), "Hand two wins!");
             assert.equal(rockPaperScissors('rock', 'scissors'), "Hand one wins!");
```
ALL TESTS ARE PASSINGGGGGGG

![dbz](https://66.media.tumblr.com/7bd02e8639eb0e4f25bc1703fe04f6bc/tumblr_nqz2xjTc1b1t1656jo1_500.gif)

Commit:
Homework Lesson 3 complete

maddiesolz commented on homework/03LessonThree.js:
```diff
@@ -0,0 +1,225 @@
+// Variable defined within a function are given local scope, meaning they can only be accessed from within the fucntion they were created in.
+// Local varables are deleted once a function is complete (usually the return value)
+// Variables declared outside of functions are automatically given global scope. Global scope variables can be accessed from anywhere within the code.
+// Variables that are created but given no value default to global scope. Even when creatd within a function.
```
STELLAR ANSWER! -insert Dragonball Z gif here-

maddiesolz commented on homework/03LessonThree.js:
```diff
@@ -0,0 +1,225 @@
+// Problem 10:
+// Replace the value 'fred' in nameArray with 'george'
+var myNewArray = nameArray.splice(1,1, 'george')
+    console.log(myNewArray);
```
Very interesting answer! You can also try nameArray[1]='george';

maddiesolz commented on homework/03LessonThree.js:
```diff
@@ -0,0 +1,225 @@
+  [4, 5, 6],
+  [7, 8, 9]
+];
+var myThreebyThreeArray = threeByThreeArray.splice(1, 1, 2);
```
See above comment

maddiesolz commented on homework/03LessonThree.js:
```diff
@@ -0,0 +1,225 @@
+var twoByTwoArray = [
+    [3, 4],
+    [7, 9]
+];
```
Yes!! I call this a 2D array. You can even have 3D arrays too, but thats a lot!

Commit:
Corrected merge issues pulled Lesson04 and sending it back up.

Commit:
TicTacToe All Tests passing, game not working 100 percent

Commit:
Homework 4 Complete

Commit:
Updated 03LessonThree

maddiesolz commented on apps/03TicTacToe.js:
```diff
@@ -1,88 +1,122 @@
+            // middle horizontal row
+            board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn ||
+            // bottom horizontal row
+            board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)
```
This is very interesting! I actually never thought to do it this way. So there is no if, and the return will just return the end  result, true/false? If possible could you show me this again during class? I would like to see it!

Awesome :)!
![leo](http://i.imgur.com/M1nEp7h.gif)

Commit:
Towers

Commit:
Towers

maddiesolz commented on homework/04LessonFour.js:
```diff
@@ -87,9 +109,39 @@ var nestedObjects = {
+// myArr[0] // -- 'manny'
+// myArr[1] // -- 2
+// myArr[2] // -- true
+// myArr[3] // -- null
```
Sounds great Manny!!! Keep it up

Commit:
05LessonFive Homework Complete

maddiesolz commented on homework/05LessonFive.js:
```diff
@@ -0,0 +1,140 @@
+
+// Problem 8:
+// Fill in forLoopArrayResult so that it matches forLoopArray
+var forLoopArrayResult = [ 0, -1, -2, -3, -4, -5, -6, -7, -8, -9 ];
```
This is correct!!! For some more practice:

``` javascript

var doubleForLoopArray = [];

for (var i=0; i <2; i++){
    for(var j=0; j<3; j++){
        doubleForLoopArray.push(i+j);
    }
}
```
1. How many elements are going to be in doubleForLoopArray?
2. What will doubleForLoopArray look like?

maddiesolz commented on apps/04TowersOfHanoi.js:
```diff
@@ -16,23 +26,85 @@ function printStacks() {
+    var endStackArray   = stacks[endStack];
+
+    if (startStackArray.length === 0) {
+      return false;
```
This is good! If i am trying to move from b to c which are both empty, this would stop me from breaking the program

maddiesolz commented on apps/04TowersOfHanoi.js:
```diff
@@ -16,23 +26,85 @@ function printStacks() {
+      return true;
+    }
+    else {
+      return startStackArray[startStackArray.length - 1] < endStackArray[endStackArray.length - 1];
```
This is good! This statement will resolve to a boolean

libersolis-dev replied:
This is very helpful. I'm going to start doing Pseudocode before I do the JS..

maddiesolz commented on apps/04TowersOfHanoi.js:
```diff
@@ -16,23 +26,85 @@ function printStacks() {
+    else {
+      console.log ('Keep Going!!!!');
+
+      return false;
```
Yes!

maddiesolz commented on apps/04TowersOfHanoi.js:
```diff
@@ -16,23 +26,85 @@ function printStacks() {
+      return true;
+    }
+    else {
+      return startStackArray[startStackArray.length - 1] < endStackArray[endStackArray.length - 1];
```
Its also helpful to draw on paper what this means. so if I follow this statements logic with my current stacks lookin like this:

```
a: [4,3,2]
b: [1]
c: []
```

im gonna try to move 2(a) onto 1(b)

startStackArray.length-1 will give me the 'top' of my stack, or '2', and endStackArray.length-1 give me the top of my other stack, or '1'...so using the statement:

2 < 1...This is False, and an illegal move. Which makes sense, cuz we cant move a bigger disk onto a smaller one.

If I was looking at this project with fresh eyes and limited experience in JavaScript syntax, I wouldn't have even thought of doing okay startStackArray.length-1...nope. What I would think is, how can I represent, in code, that this scenario: larger disk cannot be placed on smaller disk : would return a false? How can I represent with code, the 'top' of a stack, or the last element in the array?

Because the user has to follow the rules of just moving the top disks on a stack, I know I gotta check those to determine whether it will be legal or not. Here is how I would outline my isLegal function in pseudocode, then I would go back and try to do some real code by asking for syntax help:

```
function isLegal(startStack, endStack) {

//gotta find a way to relate startStack and endStack to the variable stacks in final result

if (startStack is empty){
    return illegal;
}
else if( endStack is empty) {
return legal;
}
else if(top of startStack < top of endstack){
return legal;
}
}
```

after writing that, i would try to get the correct syntax.

Commit:
Half Finished Mastermind

Commit:
pulled from upsteam

Commit:
05Mastermind.js all tests passing

Commit:
06LessonSix.js homework all tests passing and concept questions answered

Commit:
Making some suggested changes per codeacity

maddiesolz commented on apps/05Mastermind.js:
```diff
@@ -9,47 +9,118 @@ var board = [];
+
+
+
+
```
LOTS OF SPACE!

maddiesolz commented on apps/05Mastermind.js:
```diff
@@ -9,47 +9,118 @@ var board = [];
+    } // Spec 1 - this is what happens if you 'win' the game
+
+    if (guess !== solution) {
+      generateHint(solution, guess);
```
Great work Manny! Hows the understanding of this app's logic?

libersolis-dev replied:
Each time I read it, I understand a little more. Compared to 2 weeks ago I understand 1000000 percent more. 

I imagine in a few months I will be able to look back on this stuff and have a totally different way of thinking and looking at it. 

I'm starting to recognize patterns in applications, how they are laid out..

it seems like we always have 
1. a user prompt (duh, except fot that shitty game myst maybe)
2. global variables defined at the top that will be of major importance later on 
3. functions that perform the actions of the games
4. conditionals within those funcitons that present various scenarioes and results based on the data receieved 
5. Loops to make it easier to understand and less typing 
6. one main function (the game) that runs all of the functions. 
7. and alot of objects to make storing this stuff easier and for greater customization and a billonity other things.

maddiesolz commented on homework/06LessonSix.js:
```diff
@@ -37,6 +41,14 @@ var anotherPoint;
+// most important difference is that changes made to the properties of Object literals impact every new instance of that Object.
+// This is limiting.
+// Object Constructors allow specification of the properties and methods of each instance of a 'class'. Changing the age property of an object named
+// Manny from the human class from 35 to his real age of 25 will not change the age of another human created from the same class named Rob.
```
Haha very interesting!!

maddiesolz commented on homework/06LessonSix.js:
```diff
@@ -98,6 +114,12 @@ var conferenceRoom = new ConferenceRoom();
+// a class or methods can be individualized, specified, and unique so that only a certain instance of the class are able to per
+// If human is a class then walking is a method that most humans have.
+// marySol is also part of the human class. She has the walk() method at her disposal. However, because marySol is unique she has
+// certain methods other humans do not have. Each day she uses method codePearl() while most humans not only do not do this but they CANNOT do this.
```
Hahahahah!!! As is understandable...No human in their right mind should be invoking codePearl()! X_X

Manny.greatWork();

![good](http://stream1.gifsoup.com/view5/2610352/good-job-o.gif)

Commit:
mars is now complete

Commit:
Lesson7Homework 10 tests passing

Commit:
Lesson 7 Homework all tests passing

maddiesolz commented on apps/06SpaceTravelToMars.js:
```diff
@@ -9,7 +9,40 @@ var jobTypes = {
+    for (var i=0; i < this.crew.length; i++){
+      var job = this.crew[i].job;
+      if (jobTypes[job] === this.type) {
+        return this.ability;
```
Haha all tests passing! Do you have a good grasp of OOP now?

maddiesolz commented on homework/07LessonSeven/script.js:
```diff
@@ -75,18 +75,19 @@ $(document).on('ready', function() {
+  window.dataIdValue = $('#data-element').data('id');
 
   // Problem 12:
   // select the element with id="data-element"
```
Hey Manny! I am getting 2 tests failing (11 and 12) Could you resubmit?

Commit:
All tests passsinggggg

Commit:
Tictactoe Lesson07

Commit:
finally turning tictactoe07 in

Commit:
07tictactoe complete

having to do this the old fashioned way because my repository is acting strange. 

correct many issues I missed during initial submission.

Commit:
07tictactoe complete

Commit:
all tests passing

moved return to same line as code. all tests passing.

Commit:
LessonEight

Commit:
08 Homework script complete

Commit:
recommitting 07 after several issues with git

Commit:
fixed tests

Commit:
towers complete

Commit:
09todo-jquery main specs done, working on bonus specs

Commit:
all tests passing

Commit:
fixed issue with text not disapearing on submission

Commit:
app 10 address book complete

Commit:
fixing merge issues

Commit:
committing final project

mannyhagman commented:
asd

mannyhagman commented:
asdas
